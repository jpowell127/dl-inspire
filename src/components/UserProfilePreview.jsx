import React from "react";
import { Button, Card, Image } from "semantic-ui-react";

function UserProfilePreview({ user }) {
  const { first_name, last_name, username, url } = user;
  const image = user.images[50];
  return (
    <Card>
      <Card.Content>
        <Image
          floated="right"
          size="mini"
          src={image}
          alt={`Image of ${username}`}
        />
        <Card.Header>{first_name + " " + last_name}</Card.Header>
        <Card.Meta>{username}</Card.Meta>
        <Card.Description>
          <a href={url} target="_blank">
            {url}
          </a>
        </Card.Description>
      </Card.Content>
      <Card.Content extra>
        <div className="ui two buttons">
          <Button basic color="green">
            View
          </Button>
        </div>
      </Card.Content>
    </Card>
  );
}

export default UserProfilePreview;
