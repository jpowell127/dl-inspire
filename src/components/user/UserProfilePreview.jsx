import React from "react";
import PropTypes from "prop-types";
import { Button, Card, Image } from "semantic-ui-react";
import UserProfileModal from "./UserProfileModal";

export default function UserProfilePreview({ user }) {
  const { display_name, username, url, id } = user;
  const image = user.images[50];

  return (
    <Card color="pink">
      <Card.Content>
        <Image
          floated="right"
          size="mini"
          src={image}
          alt={`Image of ${username}`}
        />
        <Card.Header>{display_name}</Card.Header>
        <Card.Meta>{username}</Card.Meta>
      </Card.Content>
      <Card.Content extra>
        <UserProfileModal user={user} />
      </Card.Content>
    </Card>
  );
}

UserProfileModal.propTypes = {
  user: PropTypes.shape({
    id: PropTypes.number.isRequired,
    display_name: PropTypes.string.isRequired,
    username: PropTypes.string.isRequired,
    occupation: PropTypes.string,
    company: PropTypes.string,
    location: PropTypes.string,
    images: PropTypes.object.isRequired,
    stats: PropTypes.shape({
      followers: PropTypes.number.isRequired,
      following: PropTypes.number.isRequired,
      appreciations: PropTypes.number.isRequired,
      views: PropTypes.number.isRequired,
      comments: PropTypes.number.isRequired
    }).isRequired
  }).isRequired
};
