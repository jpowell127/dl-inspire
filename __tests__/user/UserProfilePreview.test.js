import React from "react";
import UserProfilePreview from "../../src/components/user/UserProfilePreview";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

configure({ adapter: new Adapter() });

const user = {
  display_name: "test-user",
  username: "testuser",
  url: "https://google.com",
  id: 1,
  images: {}
};

test("UserProfilePreview renders correctly", () => {
  const component = shallow(<UserProfilePreview user={user} />);
  expect(toJson(component)).toMatchSnapshot();
});
