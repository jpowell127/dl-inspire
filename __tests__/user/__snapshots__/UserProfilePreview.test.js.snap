// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfilePreview renders correctly 1`] = `
<Card
  color="pink"
>
  <CardContent>
    <Image
      alt="Image of testuser"
      as="img"
      floated="right"
      size="mini"
      ui={true}
    />
    <CardHeader>
      test-user
    </CardHeader>
    <CardMeta>
      testuser
    </CardMeta>
  </CardContent>
  <CardContent
    extra={true}
  >
    <UserProfileModal
      user={
        Object {
          "display_name": "test-user",
          "id": 1,
          "images": Object {},
          "url": "https://google.com",
          "username": "testuser",
        }
      }
    />
  </CardContent>
</Card>
`;
